packages/php/blueprint/src/ImportStep.php

```diff
@@ -80,33 +82,80 @@ function ( $key, $importer ) {
 	public function import() {
 		$result = StepProcessorResult::success( $this->step_definition->step );
 
-		if ( ! isset( $this->indexed_importers[ $this->step_definition->step ] ) ) {
-			$result->add_warn( "Unable to find an importer for {$this->step_definition->step}" );
+		if ( ! $this->can_import( $result ) ) {
 			return $result;
 		}
 
 		$importer = $this->indexed_importers[ $this->step_definition->step ];
-		// validate importer is a step processor before processing.
+
+		$logger = new Logger();
+		$logger->log(
+			sprintf( 'Starting import "%s" step', $this->step_definition->step ),
+			WC_Log_Levels::INFO,
+			array(
+				'importer' => get_class( $importer ),
+			)
+		);
+
+		$importer_result = $importer->process( $this->step_definition );
+
+		if ( $importer_result->is_success() ) {
+			$logger->log(
+				sprintf( 'Import "%s" step completed', $this->step_definition->step ),
+				WC_Log_Levels::INFO,
+				array(
+					'messages' => $importer_result->get_messages( 'info' ),
+				)
+			);
+		} else {
+			$logger->log(
+				sprintf( 'Import "%s" step failed', $this->step_definition->step ),
+				WC_Log_Levels::ERROR,
+				array(
+					'messages' => $importer_result->get_messages( 'error' ),
+				)
+			);
+		}
+
+		$result->merge_messages( $importer_result );
+
+		return $result;
+	}
+
+	/**
+	 * Check if the step can be imported.
+	 *
+	 * @param StepProcessorResult $result The result object to add messages to.
+	 *
+	 * @return bool True if the step can be imported, false otherwise.
+	 */
+	protected function can_import( &$result ) {
```

Comment:
Should we log these errors as well, if the user can't import for example?

Reply:
I think additional error logging is unnecessary, as users receive error messages in UI or CLI with `--show-messages` when imports fail, which should be a better experience. Logging is the primary focus of security-related events.

Reply:
Ah, there is a known issue that errors are not shown correctly. It's fixed in https://github.com/woocommerce/woocommerce/pull/57336.